### PR TITLE
Blocks: Show the Launchpad on save modal inside the Site Editor for all flows

### LINKED ISSUE #71717: LaunchPad - Site Editor Save Modal - Adapt for non-LIB flows.
Recently we added a launchpad modal on-save in the Site Editor that appears for link-in-bio sites. https://github.com/Automattic/jetpack/pull/27976

We should iterate on this and update it to appear for all launchpad flows. We should also ensure the text shown is relevant to the flow at hand. A general copy that fits all non-LIB cases would make sense, a copy specific for newsletters could be nice but not necessary.

To be clear - this does not mean extending this functionality into the post editor. We want this modal to remain in the site editor and just ensure all launchpad flows receive this functionality there.

### PR DESCRIPTION
**Pull request by sharpirate**
Fixes [LaunchPad - Site Editor Save Modal - Adapt for non-LIB flows](https://github.com/Automattic/wp-calypso/issues/71717) and [LaunchPad - editor 'back-to-launchpad' modal tracking events](https://github.com/Automattic/wp-calypso/issues/72097)

## Proposed changes:
This PR provides a generic modal for all Launchpad flows to allow the user to go back to the Launchpad screen from the Site Editor once they have saved their progress. It also adds tracking events

image

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion
paYE8P-1Kg-p2

## Does this pull request change what data or activity we track or use?
No

### Testing on WPCOM
1. Sandbox `public-api.wordpress.com` along with a Launchpad enabled site
2. Run `bin/jetpack-downloader test jetpack add/launchpad_save_modal_all_flows` on your sandbox
3. Go to the Site Editor
4. Make a change and save. You should see the modal appear.
5. Test with different flows - `newsletter`, `link-in-bio`, `free`, `videopress`.
6. Use [Tracks Vig ](https://github.com/Automattic/tracks-chrome-extension) to test the tracks events:
• Clicking the close button -> `jetpack_launchpad_save_modal_close`
• Clicking the Back to Edit -> `jetpack_launchpad_save_modal_back_to_edit`
• Clicking the Next Button -> `jetpack_launchpad_save_modal_next_steps`

### CODE REVIEW

File: projects/plugins/jetpack/extensions/plugins/launchpad-save-modal/index.js
Comment (jeherve) [inline projects/plugins/jetpack/extensions/plugins/launchpad-save-modal/index.js]: I think that's a typo? I don't think `Jetpack_Launchpad_save_modalSaveModal` exists.

Comment (sharpirate) [reply projects/plugins/jetpack/extensions/plugins/launchpad-save-modal/index.js]: Good catch, updated

File: projects/plugins/jetpack/extensions/plugins/launchpad-save-modal/util.js
Comment (Addison-Stavlo) [inline projects/plugins/jetpack/extensions/plugins/launchpad-save-modal/util.js]: We should probably remove this since we want to deliver this functionality for all launchpad flows, so simply checking the launchpad enabled status should be enough.

If there are other concerns im missing, we may want to add in `write` and `build` in prep of our current work to enable launchpads for those flows as well.

**Review by Addison-Stavlo — APPROVED** [review]
I tested this with various flows: free, newsletter, and link-in-bio and it works as expected. It appears in the site editor on save and not the post editor. 👍

File: projects/plugins/jetpack/extensions/plugins/launchpad-save-modal/index.js
Comment (Addison-Stavlo) [inline projects/plugins/jetpack/extensions/plugins/launchpad-save-modal/index.js]: changing from `href` to `window.location.assign` seems to trap us in the calypso site editor iframe.

Before, when exiting we properly broke out of the calypso iframe. Now on exiting the editor via the modal, notice the URL at the top bar of the browser doesnt change. You will see `*/site-editor/*` instead of the launchpad URL

File: projects/plugins/jetpack/extensions/plugins/launchpad-save-modal/index.js
Comment (Addison-Stavlo) [inline projects/plugins/jetpack/extensions/plugins/launchpad-save-modal/index.js]: Also, I noticed above where we get `launchPadUrl` it is still getting the link-in-bio URL. So my free site that I entered the site editor from a free flow checklist, the modal made me exit to view a link-in-bio checklist.

**Review by Addison-Stavlo — CHANGES_REQUESTED** [review]
Upon further inspection and testing I came across 2 issues that should be easily resolvable (more below in comments):

1. We arent updating the LaunchPadUrl based on the site_intent, so non-LIB sites using this modal are directed to the link-in-bio checklist.
2. The new method of navigation keeps us trapped in the calypso iframe.

File: projects/plugins/jetpack/extensions/plugins/launchpad-save-modal/index.js
Comment (Addison-Stavlo) [inline projects/plugins/jetpack/extensions/plugins/launchpad-save-modal/index.js:28]: nitpick, and probably not something to block this PR on. But it looks like we could set this up in a way where we don't need to re-register server side for each new intent we want to enable (see the `calypso-edit-post` example shown at the bottom of https://github.com/Automattic/jetpack/tree/trunk/projects/js-packages/components/tools/jp-redirect)

that said, im fine with that being moved to a follow up if we need to worry about it. right now we have all these source strings registered for what we want and it isnt hard to add them.

**Review by Addison-Stavlo — APPROVED** [review]
I tested this again after the recent changes and it is working as expected.

**Review by jeherve — COMMENTED** [review]
Should I wait to test this until #28555 is merged, and this PR updated?

**Comment by sharpirate** [comment]
@jeherve I think we should merge this one first since the other one builds on top of this one

**Comment by sharpirate** [comment]
@jeherve Actually we can close this one since the other one has all of the changes here

**Pull request closed without merging**
